homeassistant/components/vicare/sensor.py

```diff
@@ -698,6 +698,42 @@ class ViCareSensorEntityDescription(SensorEntityDescription, ViCareRequiredKeysM
         value_getter=lambda api: api.getPhotovoltaicProductionCumulatedCurrentDay(),
         unit_getter=lambda api: api.getPhotovoltaicProductionCumulatedUnit(),
     ),
+    ViCareSensorEntityDescription(
+        key="photovoltaic_energy_production_this_week",
+        translation_key="photovoltaic_energy_production_this_week",
+        native_unit_of_measurement=UnitOfEnergy.WATT_HOUR,
+        state_class=SensorStateClass.TOTAL_INCREASING,
+        value_getter=lambda api: api.getPhotovoltaicProductionCumulatedCurrentWeek(),
+        unit_getter=lambda api: api.getPhotovoltaicProductionCumulatedUnit(),
+        entity_registry_enabled_default=False,
+    ),
+    ViCareSensorEntityDescription(
+        key="photovoltaic_energy_production_this_month",
+        translation_key="photovoltaic_energy_production_this_month",
+        native_unit_of_measurement=UnitOfEnergy.WATT_HOUR,
+        state_class=SensorStateClass.TOTAL_INCREASING,
+        value_getter=lambda api: api.getPhotovoltaicProductionCumulatedCurrentMonth(),
+        unit_getter=lambda api: api.getPhotovoltaicProductionCumulatedUnit(),
+        entity_registry_enabled_default=False,
+    ),
+    ViCareSensorEntityDescription(
+        key="photovoltaic_energy_production_this_year",
+        translation_key="photovoltaic_energy_production_this_year",
+        native_unit_of_measurement=UnitOfEnergy.WATT_HOUR,
+        state_class=SensorStateClass.TOTAL_INCREASING,
+        value_getter=lambda api: api.getPhotovoltaicProductionCumulatedCurrentYear(),
+        unit_getter=lambda api: api.getPhotovoltaicProductionCumulatedUnit(),
+        entity_registry_enabled_default=False,
+    ),
+    ViCareSensorEntityDescription(
+        key="photovoltaic_energy_production_total",
+        translation_key="photovoltaic_energy_production_total",
+        native_unit_of_measurement=UnitOfEnergy.WATT_HOUR,
```

Comment:
Considering this being a total, it might make sense to set the suggested unit of measurement to KWh?

Reply:
Good point, I simply didn't want to flood the users with a bunch of new sensors.

Reply:
So we would need to convert here on the fly, right?

Doesn't the energy dashboard do that in the end to have better readable values?

Reply:
> So we would need to convert here on the fly, right?

Well, Home Assistant handles that. Just set the suggested unit of measurement property additionally to the native unit of measurement.

> Doesn't the energy dashboard do that in the end to have better readable values?

No, this won't matter for the energy dashboard at all. This only affects the display of sensors.

Reply:
Ok, nice.

```suggestion
        native_unit_of_measurement=UnitOfEnergy.WATT_HOUR,
        suggested_unit_of_measurement=UnitOfEnergy.KILO_WATT_HOUR,
```